liesel/model/nodes.py

```diff
@@ -976,5 +951,85 @@ def add_group(name: str, **kwargs: Node | Var) -> None:
         as keywords.
     """
 
-    for key, arg in kwargs.items():
-        arg.groups.add((name, key))
+    return Group(name, **kwargs)
+
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Group ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+class Group:
+    def __init__(self, name: str, **nodes_and_vars: Node | Var) -> None:
+        self._name = name
+        self._nodes_and_vars = nodes_and_vars
+
+        for member in self._nodes_and_vars.values():
+            if name in member.groups:
+                raise RuntimeError(
+                    f"Node is already a member of a group with the name {name}."
+                )
+            member._groups[name] = self
+
+    @property
+    def name(self) -> str:
+        """The group's name."""
+        return self._name
+
+    @property
+    def vars(self) -> MappingProxyType[str, Var]:
+        """A mapping of the variables in the group with their names as keys."""
+
+        vars_ = {
+            name: obj
+            for name, obj in self._nodes_and_vars.items()
+            if isinstance(obj, Var)
+        }
+        return MappingProxyType(vars_)
+
+    @property
+    def nodes(self) -> MappingProxyType[str, Node]:
+        """A mapping of the nodes in the group with their names as keys."""
+        nodes = {
+            name: obj
+            for name, obj in self._nodes_and_vars.items()
+            if isinstance(obj, Node)
+        }
+        return MappingProxyType(nodes)
+
+    @property
+    def nodes_and_vars(self) -> MappingProxyType[str, Node | Var]:
+        """A mapping of all group members with their names as keys."""
+        return MappingProxyType(self._nodes_and_vars)
+
+    def get_value(self, model_state: dict, name: str) -> Array:
```

Comment:
Can we find a more descriptive name here? Maybe `value_from_model_state`?

Reply:
I would prefer `get_value` over `value_from_model_state`, because 1) I think keeping this method name short is important 2) I think it is already quite clear from the argument name and docstring that the value is obtained from the model state.

Would you like something like `Group.value_from(model_state, name)` better?

Reply:
We are going with `.value_from` now.